public/app/features/playlist/api.ts

```diff
@@ -1,214 +0,0 @@
-  }
-
-  async updatePlaylist(playlist: Playlist): Promise<void> {
-    await withErrorHandling(() => getBackendSrv().put(`/api/playlists/${playlist.uid}`, playlist));
-  }
-
-  async deletePlaylist(uid: string): Promise<void> {
-    await withErrorHandling(() => getBackendSrv().delete(`/api/playlists/${uid}`), 'Playlist deleted');
-  }
-}
-
-interface PlaylistSpec {
-  title: string;
-  interval: string;
-  items: PlaylistItem[];
-}
-
-type K8sPlaylist = Resource<PlaylistSpec>;
-
-class K8sAPI implements PlaylistAPI {
-  readonly server: ResourceClient<PlaylistSpec>;
-
-  constructor() {
-    this.server = new ScopedResourceClient<PlaylistSpec>({
-      group: 'playlist.grafana.app',
-      version: 'v0alpha1',
-      resource: 'playlists',
-    });
-  }
-
-  async getAllPlaylist(): Promise<Playlist[]> {
-    const result = await this.server.list();
-    return result.items.map(k8sResourceAsPlaylist);
-  }
-
-  async getPlaylist(uid: string): Promise<Playlist> {
-    const r = await this.server.get(uid);
-    const p = k8sResourceAsPlaylist(r);
-    await migrateInternalIDs(p);
-    return p;
-  }
-
-  async createPlaylist(playlist: Playlist): Promise<void> {
-    const body = this.playlistAsK8sResource(playlist);
-    await withErrorHandling(async () => {
-      await this.server.create(body);
-    });
-  }
-
-  async updatePlaylist(playlist: Playlist): Promise<void> {
-    const body = this.playlistAsK8sResource(playlist);
-    await withErrorHandling(() => this.server.update(body).then(() => {}));
-  }
-
-  async deletePlaylist(uid: string): Promise<void> {
-    await withErrorHandling(() => this.server.delete(uid).then(() => {}), 'Playlist deleted');
-  }
-
-  playlistAsK8sResource = (playlist: Playlist): ResourceForCreate<PlaylistSpec> => {
-    return {
-      metadata: {
-        name: playlist.uid, // uid as k8s name
-      },
-      spec: {
-        title: playlist.name, // name becomes title
-        interval: playlist.interval,
-        items: playlist.items ?? [],
-      },
-    };
-  };
-}
-
-// This converts a saved k8s resource into a playlist object
-// the main difference is that k8s uses metadata.name as the uid
-// to avoid future confusion, the display name is now called "title"
-function k8sResourceAsPlaylist(r: K8sPlaylist): Playlist {
-  const { spec, metadata } = r;
-  return {
-    uid: metadata.name, // use the k8s name as uid
-    name: spec.title,
-    interval: spec.interval,
-    items: spec.items,
-  };
-}
-
-/** @deprecated -- this migrates playlists saved with internal ids to uid  */
-async function migrateInternalIDs(playlist: Playlist) {
-  if (playlist?.items) {
-    for (const item of playlist.items) {
-      if (item.type === 'dashboard_by_id') {
-        item.type = 'dashboard_by_uid';
-        const uids = await getBackendSrv().get<string[]>(`/api/dashboards/ids/${item.value}`);
-        if (uids?.length) {
-          item.value = uids[0];
-        }
-      }
-    }
-  }
-}
```

Comment:
do we still need this? i've removed it, but can add it back if needed

Reply:
@ryantxu sorry for direct ping - do you happen to know if this is still needed? 🙏

Reply:
If easy to keep... can we keep it?  If it is difficult, we can note it as a breaking change and that is fine